test/integration/discovery.go

```diff
@@ -279,6 +324,58 @@ var _ = Describe("Discover Cluster", func() {
 	})
 })
 
+func validateBuyerSellerCommodity(entityDTOs []*proto.EntityDTO, nodeName string, podName string, commodityType proto.CommodityDTO_CommodityType) bool {
+
+	var buyerRegsitered bool
+	var sellerRegistered bool
+	var podEntityDTO *proto.EntityDTO
+	var nodeEntityDTO *proto.EntityDTO
+
+	for _, entityDTO := range entityDTOs {
+		if *entityDTO.EntityType == proto.EntityDTO_CONTAINER_POD &&
+			*entityDTO.DisplayName == podName {
+			podEntityDTO = entityDTO
+			buyerRegsitered = true
+
+		} else if *entityDTO.EntityType == proto.EntityDTO_VIRTUAL_MACHINE &&
```

Comment:
Thanks for doing this @NiraliKotak. This complete validation would be to check if the node(s) on which you have applied the label carries the access commodity and no other node has that commodity.

Also the whole logic here could be simpler. Look for Node dtos in a saperate section, and check its commodities in that loop. Look pod dto in another section and check its commodities (something similar to what you had earlier).

Reply:
I am not too worried about the logic or your style of code. We need to however ensure that we do check for all conditions

Reply:
@irfanurrehman Thanks for reviewing. In line 364 I am validating the particular node where the pod is hosted (which is the one where I have applied the label) has the Access commodity . Will it be okay if I update the code further verifying if the node which is labeled is the same as the node where pod(with pv) is hosted?

Reply:
@NiraliKotak, no that is not what is needed. Because as part of the test setup, you have already ensured that the pod is on this node. What I meant is check other nodes from the cluster; they should not have the access commodity with this key in the sold commodities list.